### PR TITLE
New Entries: PHP Date Functions

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Created a PHP Date Functions Entry, with a few terms for date functions. From the Target Keyword list.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/php/concepts/date-functions/terms/date/date.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Date'
+  - 'Time'
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `date()` function crates a formatted date-time string from a local date-time.
```
Comment (KTom101) [inline content/php/concepts/date-functions/terms/date/date.md:15]:
```suggestion
The `date()` function creates a formatted date-time string from a local date-time.
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@SSwiniarski A few changes.

**Comment by SSwiniarski** [comment]
@KTom101, I implemented your suggestions.

File: content/php/concepts/date-functions/terms/date-create/date-create.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `date_create()` function creates a new `DateTime` object based on a formatted date-time string passed to it, along with an optional timezone.
+
+## Syntax
+
+```php
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-create/date-create.md:20]:
```suggestion
```pseudo
```

File: content/php/concepts/date-functions/terms/date-create/date-create.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+date_create($datetime,$timezone)
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-create/date-create.md:21]:
```suggestion
date_create($datetime, $timezone)
```

File: content/php/concepts/date-functions/terms/date-create/date-create.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+| Description | Example                  |
+| ----------- | ------------------------ |
+| EXIF        | "2011:12:31 15:25:27"    |
+| MySQL       | "2011-12-31 15:25:27"    |
+| SOAP        | "2011-12-31T15:25:27.05" |
+| XMLRPC      | "20111231T15:25:27"      |
+| WDDX        | "2011-12-31T15:25:27"    |
+
+## Codebyte Example
+
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-create/date-create.md:69]: For these sections, let's add a brief sentence here that describes what's happening in the Codebyte below.

File: content/php/concepts/date-functions/terms/date-format/date-format.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+## Syntax
+
+```php
+$datestring = date_format($object,$format);
+```
+
+The `$object` parameter is a `DateTime` object as returned by [`date_create()`](https://www.codecademy.com/resources/docs/php/date-functions/date_create). And the `$format` parameter is a format string that can use the following characters:
+
+| Character | Description                                                                        |
+| --------- | ---------------------------------------------------------------------------------- |
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-format/date-format.md:27]:
```suggestion
| :---------: | ---------------------------------------------------------------------------------- |
```

File: content/php/concepts/date-functions/terms/date-format/date-format.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `date_format()` function takes a `DateTime` object and a format string and returns a string based on the formatted value.
+
+## Syntax
+
+```php
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-format/date-format.md:20]:
```suggestion
```pseudo
```

File: content/php/concepts/date-functions/terms/date-format/date-format.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+$datestring = date_format($object,$format);
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-format/date-format.md:21]:
```suggestion
$datestring = date_format($object, $format);
```

File: content/php/concepts/date-functions/terms/date-format/date-format.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+| `c`       | Full ISO 8601 date                                                                 |
+| `r`       | RFC 2822 formatted date                                                            |
+| `U`       | Seconds since the Unix Epoch (Midnight GMT January 1, 1970)                        |
+
+## Codebyte Example
+
+```codebyte/php
+<?php
+  $date = date_create("12/1/2020 1:30pm");
+  echo(date_format($date,"G:i l, F jS, Y"));
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date-format/date-format.md:67]:
```suggestion
  echo(date_format($date, "G:i l, F jS, Y"));
```

File: content/php/concepts/date-functions/terms/getdate/getdate.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+- `$t["mon"]` int representing the month. (1 to 12)
+- `$t["year"]` int representing the four digit year.
+- `$t["yday"]` int representing the day of the year. (0 to 365)
+- `$t["weekday"]` string with the full name of the weekday. ("Sunday" to "Saturday")
+- `$t["month"]` string with full name of the month. ("January" to "December")
+- `$t[0]` int representing the Unix timestamp.
+
+## Codebyte Example
+
+```php
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/getdate/getdate.md:41]:
```suggestion
```codebyte/php
```

File: content/php/concepts/date-functions/terms/time/time.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `time()` function returns the current time measured in seconds from the Unix Epoch, Midnight GMT January 1, 1970.
+
+## Codebyte Example
+
+```php
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/time/time.md:19]:
```suggestion
```codebyte/php
```

File: content/php/concepts/date-functions/terms/date/date.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `date()` function creates a formatted date-time string from a local date-time.
+
+## Syntax
+
+```php
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/date/date.md:19]:
```suggestion
```pseudo
```

File: content/php/concepts/date-functions/terms/getdate/getdate.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `getdate()` function returns an array representing information about the current date or a Unix timestamp.
+
+## Syntax
+
+```php
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/getdate/getdate.md:19]:
```suggestion
```pseudo
```

File: content/php/concepts/date-functions/terms/getdate/getdate.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+$t = getdate($timestamp);
+```
+
+The `$timestamp` parameter is optional. It is an int representing a Unix timestamp, i.e. the number of seconds from the Unix Epoch, midnight GMT on January 1st, 1970. If omitted, `getdate()` returns information on the current date and time.
+
+The `getdate()` function returns an associative array, `$t`, providing information on the current date or `$timestamp` with the following members:
+
```
Comment (Dusch4593) [inline content/php/concepts/date-functions/terms/getdate/getdate.md:26]: What do we think about turning this list into a markdown table that looks similar to this?

```
| Member | Data Type | Description | Range |
| :-------: | :---------: | ----------- | ------- |
| `$t["seconds"]` | int | Number of seconds. | 0-59 |
| `$t["weekday"]` | string | Full name of the weekday. | "January" - "December" |
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Finished with my review and left you some comments/suggestions to check out.

**Comment by SSwiniarski** [comment]
@Dusch4593, I implemented all your suggestions.

File: content/php/concepts/date-functions/date-functions.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Date'
+  - 'Time'
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+PHP manipulates dates in a number of ways, using arrays, strings, Unix timestamps, or `DateTime` objects. PHP provides a number of built-in functions manipulating dates using these various data types.
```
Comment (caupolicandiaz) [inline content/php/concepts/date-functions/date-functions.md:15]:
```suggestion
PHP manipulates dates in a number of ways, using arrays, strings, Unix timestamps, or `DateTime` objects. PHP provides a number of built-in functions for manipulating dates using these various data types.
```

File: content/php/concepts/date-functions/terms/date-create/date-create.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+```codebyte/php
+
+The below example creates `DateTime` objects using two different string formats.
```
Comment (caupolicandiaz) [inline content/php/concepts/date-functions/terms/date-create/date-create.md:72]:
```suggestion
The example below creates `DateTime` objects using two different string formats.
```

File: content/php/concepts/date-functions/terms/date/date.md [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+$datestring = date($format, $timestamp);
+```
+
+The `$format` string is a string specifying how the date/time is to be formatted. It uses the same syntax as the [`date_format()`](https://www.codecademy.com/resources/docs/php/date-functions/date-format) function.
+
+The `$timestamp` parameter is a Unix timestamp, the date-time expressed as seconds elapsed from the Unix Epoch (Midnight GMT, January 1st, 1970). It is optional, and if omitted or `NULL` it defaults to the current local date and time.
+
+## Codebyte Example
+
+The below example prints the current date and time using the `date()` function.
```
Comment (caupolicandiaz) [inline content/php/concepts/date-functions/terms/date/date.md:29]:
```suggestion
The example below prints the current date and time using the `date()` function.
```

File: content/php/concepts/date-functions/terms/date-format/date-format.md [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+|    `I`    | 1 if date is in Daylight Savings Time, 0 otherwise                                 |
+|    `O`    | Difference to GMT without colon                                                    |
+|    `P`    | Difference to GMT with colon                                                       |
+|    `c`    | Full ISO 8601 date                                                                 |
+|    `r`    | RFC 2822 formatted date                                                            |
+|    `U`    | Seconds since the Unix Epoch (Midnight GMT January 1, 1970)                        |
+
+## Codebyte Example
+
+The below example creates as `DateTime` object, then prints the date with a format string.
```
Comment (caupolicandiaz) [inline content/php/concepts/date-functions/terms/date-format/date-format.md:64]:
```suggestion
The example below creates a `DateTime` object, then prints the date with a format string.
```

File: content/php/concepts/date-functions/terms/getdate/getdate.md [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+|   `$t["mon"]`   |    int    | Number of the month.                 | 1-12                    |
+|  `$t["year"]`   |    int    | Four digit year.                     |
+|  `$t["yday"]`   |    int    | Day of the year.                     | 0-365                   |
+| `$t["weekday"]` |  string   | Full name of the weekday.            | "Sunday" to "Saturday"  |
+|  `$t["month"]`  |  string   | Full name of the month.              | "January" to "December" |
+|     `$t[0]`     |    int    | The Unix timestamp.                  |
+
+## Codebyte Example
+
+The below example gets an array from the `getdate()` function, and prints out various elements from it.
```
Comment (caupolicandiaz) [inline content/php/concepts/date-functions/terms/getdate/getdate.md:43]:
```suggestion
The example below gets an array from the `getdate()` function, and prints out various elements from it.
```

File: content/php/concepts/date-functions/terms/time/time.md [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The `time()` function returns the current time measured in seconds from the Unix Epoch, Midnight GMT January 1, 1970.
+
+## Codebyte Example
+
+The below example prints out the Unix timestamp for the current date.
```
Comment (caupolicandiaz) [inline content/php/concepts/date-functions/terms/time/time.md:19]:
```suggestion
The example below prints out the Unix timestamp for the current date.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@SSwiniarski looks pretty good, just some minor edits.

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski LGTM! 👍🏻 

After addressing @caupolicandiaz's comments we should be good to merge. 😄

**Comment by SSwiniarski** [comment]
@caupolicandiaz, @Dusch4593, done.

**Comment by SSwiniarski** [comment]
@caupolicandiaz, are we good to merge?

**Review by yangc95 — APPROVED** [review]

**Comment by SSwiniarski** [comment]
> @caupolicandiaz, are we good to merge?

Bump.

**Review by caupolicandiaz — APPROVED** [review]

**Comment by Dusch4593** [comment]
@SSwiniarski Looks good for merge! I made one small edit in 297ef053b072e029cdeda2052db7ae75ffa99bb4 where I moved the sentence in `date-create.md` out of the Codebyte snippet and placed above it.

**Comment by Dusch4593** [comment]
cc @SSwiniarski @caupolicandiaz @KTom101 @yangc95

Here are the live entry links:

- PHP Date Functions: https://www.codecademy.com/resources/docs/php/date-functions
- date_create(): https://www.codecademy.com/resources/docs/php/date-functions/date-create
- date_format(): https://www.codecademy.com/resources/docs/php/date-functions/date-format
- date(): https://www.codecademy.com/resources/docs/php/date-functions/date
- getdate(): https://www.codecademy.com/resources/docs/php/date-functions/getdate
- time(): https://www.codecademy.com/resources/docs/php/date-functions/time

If we find anything that needs updated, let's open a quick patch PR and we can review/merge the edits.

**Pull request merged**